group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupMemberTest.java

```diff
@@ -0,0 +1,307 @@
+
+public class ConsumerGroupMemberTest {
+
+    @Test
+    public void testNewMember() {
+        Uuid topicId1 = Uuid.randomUuid();
+        Uuid topicId2 = Uuid.randomUuid();
+        Uuid topicId3 = Uuid.randomUuid();
+
+        ConsumerGroupMember member = new ConsumerGroupMember.Builder("member-id")
+            .setMemberEpoch(10)
+            .setPreviousMemberEpoch(9)
+            .setNextMemberEpoch(11)
+            .setInstanceId("instance-id")
+            .setRackId("rack-id")
+            .setRebalanceTimeoutMs(5000)
+            .setClientId("client-id")
+            .setClientHost("hostname")
+            .setSubscribedTopicNames(Arrays.asList("foo", "bar"))
+            .setSubscribedTopicRegex("regex")
+            .setServerAssignorName("range")
+            .setClientAssignors(Collections.singletonList(
+                new ClientAssignor(
+                    "assignor",
+                    (byte) 0,
+                    (byte) 0,
+                    (byte) 1,
+                    new VersionedMetadata(
+                        (byte) 1,
+                        ByteBuffer.allocate(0)))))
+            .setAssignedPartitions(mkAssignment(
+                mkTopicAssignment(topicId1, 1, 2, 3)))
+            .setPartitionsPendingRevocation(mkAssignment(
+                mkTopicAssignment(topicId2, 4, 5, 6)))
+            .setPartitionsPendingAssignment(mkAssignment(
+                mkTopicAssignment(topicId3, 7, 8, 9)))
+            .build();
+
+        assertEquals("member-id", member.memberId());
+        assertEquals(10, member.memberEpoch());
+        assertEquals(9, member.previousMemberEpoch());
+        assertEquals(11, member.nextMemberEpoch());
+        assertEquals("instance-id", member.instanceId());
+        assertEquals("rack-id", member.rackId());
+        assertEquals("client-id", member.clientId());
+        assertEquals("hostname", member.clientHost());
+        // Names are sorted.
+        assertEquals(Arrays.asList("bar", "foo"), member.subscribedTopicNames());
+        assertEquals("regex", member.subscribedTopicRegex());
+        assertEquals("range", member.serverAssignorName().get());
+        assertEquals(
+            Collections.singletonList(
+                new ClientAssignor(
+                    "assignor",
+                    (byte) 0,
+                    (byte) 0,
+                    (byte) 1,
+                    new VersionedMetadata(
+                        (byte) 1,
+                        ByteBuffer.allocate(0)))),
+            member.clientAssignors());
+        assertEquals(mkAssignment(mkTopicAssignment(topicId1, 1, 2, 3)), member.assignedPartitions());
+        assertEquals(mkAssignment(mkTopicAssignment(topicId2, 4, 5, 6)), member.partitionsPendingRevocation());
+        assertEquals(mkAssignment(mkTopicAssignment(topicId3, 7, 8, 9)), member.partitionsPendingAssignment());
+    }
+
+    @Test
+    public void testEquals() {
+        Uuid topicId1 = Uuid.randomUuid();
```

Comment:
Is there a reason why we are initialising the topicIds in every test method instead of just once in the class?

Reply:
I personally don't like having all those global attributes defined at the class level in tests. Overtime those get messy. You never know what is defined there when you read individual tests so I usually prefer to define what I need in the test itself.

Reply:
Makes sense, but since the id is a randomUuid anyways and the name of the attribute is self explanatory might be fine no?

Reply:
I have change it a bit. Let me know what you think.

Reply:
If we had a static class and used the same IDs, wouldn't they collide?

Reply:
What do you mean by static class? Are you referring to a static attribute for the context in the test class that is shared by all the tests? In this case, yes, they would collide.

Reply:
Yes I was referring to the context.

Reply:
Sorry -- my bad for misunderstanding Java earlier 😅